pkg/kube/wait.go

```diff
@@ -70,10 +73,18 @@ func (w *waiter) waitForDeletedResources(deleted ResourceList) error {
 
 	return wait.PollUntilContextCancel(ctx, 2*time.Second, true, func(ctx context.Context) (bool, error) {
 		for _, v := range deleted {
+			for _, resource := range w.c.readyResources {
+				if resource == v {
+					continue
+				}
+			}
+
 			err := v.Get()
 			if err == nil || !apierrors.IsNotFound(err) {
 				return false, err
 			}
+			w.c.readyResources = append(w.c.readyResources, v)
+			w.log("Deleted %s %s/%s", v.Mapping.GroupVersionKind.Kind, v.Namespace, v.Name)
```

Comment:
Using the key from the helper will get you this along with the group which is important since anyone can have the same kind. i.e. rancher has at least 4 node types under different groups.